demos/smart_classroom_demo_gapi/include/custom_kernels.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "action_detector.hpp"
+#include "face_reid.hpp"
+#include "recognizer.hpp"
+#include "tracker.hpp"
+
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+
+namespace state {
+    /** NOTE: Questionable solution
+      * kernels use num_frames,
+      * num_frames total state need for FPS counter
+      */
+    extern size_t work_num_frames, total_num_frames;
+}
+
+template<> struct cv::detail::CompileArgTag<TrackerParams> {
+    static const char* tag() {
+        return "custom.get_action_detection_result_state_params";
+    }
+};
+
+template<> struct cv::detail::CompileArgTag<TrackerParamsPack> {
+    static const char* tag() {
+        return "custom.get_recognition_result_state_params";
+    }
+};
+
+namespace custom {
+    G_API_OP( FaceDetectorPostProc
+            , <cv::GArray<detection::DetectedObject>( cv::GMat
+                                                    , cv::GMat
+                                                    , detection::FaceDetectionKernelInput)>
```

Comment:
Where `detection::FaceDetectionKernelInput` is defined? Does it makes sense to define it here, closer to the operations using it?